engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java

```diff
@@ -143,6 +143,8 @@
    * value: list of nodes which are in oio with oio owning thread node
    */
   protected final Map<Integer, ArrayList<Integer>> oioGroups = new ConcurrentHashMap<>();
+  //a simple map which maps the oio node to it's the node which owns the thread.
+  protected final Map<Integer, Integer> oioNodes = new ConcurrentHashMap<>();
```

Comment:
As you have made it global make sure that the entries are consistent with oioGroups. i.e the parent to children mapping and child to parent mapping is always consistent in both the structures.